0005-configuration-management-extension/README.md

```diff
@@ -0,0 +1,163 @@
+---
+name: Configuration Management Extension
+shortname: 5/CMX
+status: raw
+editor: Dmitry Sergeev <dsergeev@cybervisiontech.com>
+---
+
+## Introduction
+
+The Configuration Management Extension is a [Kaa Protocol](/0001-kaa-protocol/README.md) extension.
+
+It is intended to manage endpoint configuration distribution.
+
+## Requirements and constraints
+### Problems and possible solutions
+
+1. _Multiple configurations available for client._ It is possible that there will be more than one available configurations for endpoint (e.g., endpoint was offline while new configurations were added).
+   
+   Solution:
+   - Only latest configuration should be sent to an endpoint to reduce amount of traffic.
+
+2. _The server should know if a configuration has been delivered._ 
+
+   Solutions:
+   - Send PUBLISH message from client to "/status" topic. See below sections for details.
+
+## Use cases
+
+### UC1
+Configuration delivery by request. The endpoint should be able to receive latest configuration from CMX by request.
+
+### UC2
+Configuration delivery that is initiated by the server. The endpoint should receive latest configuration when it connects to a server if this configuration hadn't applied yet.
+
+## Design
+
+### Request/response
+The extension uses request/response pattern. A response from client is sent to confirm delivery.
+
+For MQTT, responses MUST be published at `<request_path>/status`. Each response MUST be published with the same QoS as the corresponding request.
```

Comment:
I've defined the req/resp pattern more specifically in https://github.com/kaaproject/kaa-rfcs/pull/24. It might be enough to just link to that.
I think I'll move the QoS mention to RFC1.